components/menu/index.css

```diff
@@ -10,170 +10,637 @@ OF ANY KIND, either express or implied. See the License for the specific languag
-        padding-inline-end: calc(
-          var(--spectrum-listitem-texticon-padding-right) -
-            var(--spectrum-popover-border-size)
-        );
-      }
+    .spectrum-Menu-itemDescription {
+      line-height: var(--mod-menu-item-description-line-height-cjk, var(--spectrum-menu-item-description-line-height-cjk));
+    }
+
+    .spectrum-Menu-sectionHeading {
+      line-height: var(--mod-menu-section-header-line-height-cjk, var(--spectrum-menu-section-header-line-height-cjk));
     }
   }
+
   .spectrum-Menu-divider {
     overflow: visible;
-
     inline-size: auto;
-    margin-block: calc(var(--spectrum-listitem-texticon-divider-padding) / 2);
-    margin-inline: var(--spectrum-listitem-texticon-padding-y);
+    margin-block: var(--mod-menu-item-bottom-edge-to-text, var(--spectrum-menu-item-bottom-edge-to-text));
+  }
+}
+
+/* menus with "selectable" menuitems */
+.spectrum-Menu.is-selectable {
+  /* provide inline-start spacing for potential checkmarks */
+  .spectrum-Menu-item {
+    --spectrum-menu-checkmark-display: var(--spectrum-menu-checkmark-display-hidden);
+    padding-inline-start: var(--mod-menu-item-selectable-edge-to-text-not-selected, var(--spectrum-menu-item-selectable-edge-to-text-not-selected));
   }
+
+  /* remove inline-start spacing once an item is selected and a checkmark appears */
+  .spectrum-Menu-item--is-selected {
+    --spectrum-menu-checkmark-display: var(--spectrum-menu-checkmark-display-shown);
+    padding-inline-start: var(--mod-menu-item-label-inline-edge-to-content, var(--spectrum-menu-item-label-inline-edge-to-content));
+}
+}
+
+/* menus with "selectable" menuitems */
+/* provide inline-start spacing for potential checkmarks */
+/* .spectrum-Menu-item--is-selectable {
+  --spectrum-menu-checkmark-display: var(--spectrum-menu-checkmark-display-hidden);
+
+  padding-inline-start: var(--mod-menu-item-selectable-edge-to-text-not-selected, var(--spectrum-menu-item-selectable-edge-to-text-not-selected));  
+} */
+
+/* remove inline-start spacing once an item is selected and a checkmark appears */
+/* .spectrum-Menu-item--is-selected {
+  --spectrum-menu-checkmark-display: var(--spectrum-menu-checkmark-display-shown);
+
+  padding-inline-start: var(--mod-menu-item-label-inline-edge-to-content, var(--spectrum-menu-item-label-inline-edge-to-content));
+} */
+
+.spectrum-Menu-itemIcon {
+  fill: var(--highcontrast-menu-item-label-icon-color-default, var(--mod-menu-item-label-icon-color-default, var(--spectrum-menu-item-label-icon-color-default)));
 }
 
 .spectrum-Menu-checkmark {
-  transform: scale(1);
+  align-self: center;
+  display: var(--mod-menu-checkmark-display, var(--spectrum-menu-checkmark-display));
+  fill: var(--highcontrast-menu-checkmark-icon-color-default, var(--mod-menu-checkmark-icon-color-default, var(--spectrum-menu-checkmark-icon-color-default)));
   opacity: 1;
 }
 
-.spectrum-Menu-item {
+.spectrum-Menu-item,
+.spectrum-Menu li:not(.spectrum-Menu-item, .spectrum-Menu-divider) {
```

Comment:
Should `:not(.spectrum-Menu-item)` really be getting the same styles as `.spectrum-Menu-item` in this case?

Reply:
The rules here for `padding-block-start` and `end` make sense, but I'm not sure about the other rules. It's adding the inline-padding twice if you have a menu heading.